dev/conductor/core/lib/src/packages_autoroller.dart

```diff
@@ -0,0 +1,263 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:convert';
+import 'dart:io' as io;
+
+import 'package:process/process.dart';
+
+import 'git.dart';
+import 'globals.dart';
+import 'repository.dart';
+
+/// A service for rolling the SDK's pub packages to latest and open a PR upstream.
+class PackageAutoroller {
+  PackageAutoroller({
+    required this.githubClient,
+    required this.token,
+    required this.framework,
+    required this.orgName,
+    required this.processManager,
+  }) {
+    if (token.trim().isEmpty) {
+      throw Exception('empty token!');
+    }
+    if (githubClient.trim().isEmpty) {
+      throw Exception('Must provide path to GitHub client!');
+    }
+    if (orgName.trim().isEmpty) {
+      throw Exception('Must provide an orgName!');
+    }
+  }
+
+  final FrameworkRepository framework;
+  final ProcessManager processManager;
+
+  /// Path to GitHub CLI client.
+  final String githubClient;
+
+  /// GitHub API access token.
+  final String token;
+
+  static const String hostname = 'github.com';
+
+  static const String prBody = '''
+This PR was generated by `flutter update-packages --force-upgrade`.
```

Comment:
Is Flutter update-packages going to call this conductor logic?

Reply:
The flow will be:

luci recipe -> dart packages_autoroller.dart -> flutter update-packages --force-upgrade

Reply:
I'm using the conductor codebase because it has abstractions for cloning the flutter/flutter repo in a somewhat hermetic environment, and there's both unit test and integration test infrastructure already set up.